### PR TITLE
feat: [CDS-56694]:Rancher infrastructure

### PR DESCRIPTION
**Pull request by litvinovstas**
### Summary
Added rancher infrastructure

#### Screenshots
Screenshot 2023-06-20 at 22 57 53
Screenshot 2023-06-20 at 22 58 26
Screenshot 2023-06-20 at 22 58 55
Screenshot 2023-06-21 at 00 10 14

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by litvinovstas** [comment]
retrigger jest

File: src/modules/75-cd/components/PipelineSteps/RancherInfrastructureSpec/__tests__/RancherInfraStructureSpec.test.tsx [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+          environmentRef: 'environmentRef',
+          infrastructureRef: 'infrastructureRef'
+        }}
+        type={StepType.KubernetesRancher}
+        stepViewType={StepViewType.InputSet}
+        onUpdate={onUpdateHandler}
+      />
+    )
+
+    fireEvent.click(getByText(container, 'Submit'))
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineSteps/RancherInfrastructureSpec/__tests__/RancherInfraStructureSpec.test.tsx:180]: Please rebase with develop and fix this

**Review by bhavya-sinha — CHANGES_REQUESTED** [review]

**Comment by litvinovstas** [comment]
retrigger cypress-cv

**Comment by litvinovstas** [comment]
retrigger cypress-rest

File: src/modules/75-cd/components/PipelineSteps/RancherInfrastructureSpec/RancherInfrastructureSpec.tsx [hunk 2]
```diff
@@ -0,0 +1,592 @@
...
+
+import pipelineVariableCss from '@pipeline/components/PipelineStudio/PipelineVariables/PipelineVariables.module.scss'
+
+const logger = loggerFor(ModuleName.CD)
+
+interface RancherListResponse extends RancherListClustersResponseResponse {
+  clusters: string[]
+}
+
+type CommonConnector = 'Rancher'
```
Comment (Sagar-Rajput7) [inline src/modules/75-cd/components/PipelineSteps/RancherInfrastructureSpec/RancherInfrastructureSpec.tsx:71]: NIT: can use connector enum here

File: src/modules/75-cd/components/PipelineSteps/RancherInfrastructureSpec/RancherInfrastructureSpec.tsx [hunk 3]
```diff
@@ -0,0 +1,592 @@
...
+              />
+            </FormikForm>
+          )
+        }}
+      </Formik>
+    </Layout.Vertical>
+  )
+}
+
+const RancherInfrastructureSpecInputForm: React.FC<RancherInfrastructureSpecEditableProps & { path: string }> = ({
```
Comment (Sagar-Rajput7) [inline src/modules/75-cd/components/PipelineSteps/RancherInfrastructureSpec/RancherInfrastructureSpec.tsx:212]: NIT: could split these into maybe 3 files, Edit, inputset, and parent

File: src/modules/75-cd/components/PipelineSteps/RancherInfrastructureSpec/__tests__/RancherInfraStructureSpec.test.tsx [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+      <TestStepWidget
+        initialValues={initialValues}
+        template={runtimeInputsValues}
+        allValues={initialValues}
+        type={StepType.KubernetesRancher}
+        stepViewType={StepViewType.InputVariable}
+      />
+    )
+
+    expect(container).toBeDefined()
```
Comment (Sagar-Rajput7) [inline src/modules/75-cd/components/PipelineSteps/RancherInfrastructureSpec/__tests__/RancherInfraStructureSpec.test.tsx:133]: I see multiple occurrences of `expect(container).toBeDefined()`, we are not asserting any useful info here.
Maybe here we can assert what the test title says, for e.g. we can assert the visibility of input or other fields with there default states e.g. whether its fixed/runtime/expression, etc.

File: src/modules/75-cd/components/PipelineSteps/RancherInfrastructureSpec/RancherInfrastructureSpec.tsx (hunk 3 above)
Comment (litvinovstas) [reply src/modules/75-cd/components/PipelineSteps/RancherInfrastructureSpec/RancherInfrastructureSpec.tsx:212]: we are using same pattern on project level.

**Review by Sagar-Rajput7 — APPROVED** [review]

**Review by reetika04 — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Pull request merged**